### PR TITLE
[HFSM] Add telemetry documentation

### PR DESCRIPTION
**Pull request by vaavva**
Added:
- information about telemetry samples
- telemetry uml diagrams
- updated rpc uml diagrams that were out of date
- minor readme fixes

### CODE REVIEW

File: sdk/docs/core/az_mqtt_rpc.md [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
...edYxcyt0QnQlmqsXmVxtixL_DrlqZGoOwPNbDjAxBod0fz8LTVZ-6ZNWIyaK0FgCm6fX2oALwOIh337mkA0UkG2g-vO8Lbh72b016WDpQ572Ho5fareNISz_1s8B_C8PP3glbD9xUpeZDAfnSVOvdyojR3YtE3B128SjU8K4Ke6tyqRxxUlCp9GHSmxPcYWc3c86He51NqWPyiV5R4eDXHaJfb-lrNLkDSBLQh81JwH5qdUXKyGGjt1yrQWE6bur2HFx7yI2aZLaK8AInd1rZA1kTcG0fKeHQ5AYrYv0Md1ngqONIH6YBHkzlPXJkfp6mgefJLXG9Ivz6X9t5RMNRrQwuJgDQDxKW2t1X9IiPgK9x5DoYZRhCAbdmz1LIMHGxxPWz9j2yIFNawadxprYyZZrF6sOGLbKwRHBusmDezcPPWHwuVt01xvFBlHBDvlUBcv04Etz_dbnQK5ChuLFvoJQvdLt5wsWIXS5R29ivATu1z7BskwRUymkyZjargmk3cGUxhmOTXL6l3gvNNl-mAOB3WptjxnRt_oRs81xClZZ-e6im8Z-5fNp6Gnbo9Si1IUX0nRJUcKfxMBq2yGKZYR_TxZHEqE8WddsLycJ-6Lnc9yW0q1LmtROZ5riFzbU7XIH_rLmMDUZv9pSxHraD-nrl7Ih4KTh7i7ejrA9CVeqNDm5Gj_pstU_mxhBNoE_i-WH40ZmVQigk6VVmLZJM6YJditG-qtiQ8MsE3PIHiIBVcEVsk-9fN3AJFgBs3nCbs9xGhZRHx2vSR2A2xFiFqWZlsuGcqnjw47oPGoeL3NJWMJ1mKOPcOF0WtiphbAmSA-9toXXu4AVCWLxg_o-q1qVLXFqQkbo3hebv3dx5b2PEkavJn4mtW-5WXDX0ct1DB_pTyZQqJTxulAd4UkayiD1u1ErwihcDmglZO3J_2B0JvXkSVyEyIS7JNgFk_B6fnB8iJcLPCf4vgWEzGttHvJuW2M7E8h-g4xfj57duWYwx2-eYzv06tNfG_SdmDSA-DV2-UDr3tntkFOqdlPv26eH46YDBa3bqa1x0-JbCdC2Gry9zQ0_-QqourQZvJHg01-ut203FeJ71Xmz0KaMBvMmQj5O0evZGd6B45QBoNKs2ndZ2N6iTO_PN_9ZhCdOa7GU_OU8SldZqPqr0MNiVxLTT5ly0 "mqtt_rpc_client")
```
Comment (RLeclair) [inline sdk/docs/core/az_mqtt_rpc.md:5]: Don't forget to update from your other PR https://github.com/Azure/azure-sdk-for-c/pull/2763

Comment (RLeclair) [reply sdk/docs/core/az_mqtt_rpc.md:5]: Same for the one below

**Review by RLeclair — APPROVED** [review]
LGTM!

File: sdk/docs/core/az_mqtt_rpc.md (hunk 1 above)
Comment (vaavva) [reply sdk/docs/core/az_mqtt_rpc.md]: done!

**Pull request merged**